### PR TITLE
Thoms' changes to architecture section

### PR DESCRIPTION
**Pull request by AnthonyLim23**
@thomasholmrod changes to the architecture section

### CODE REVIEW

File: guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex [hunk 1]
```diff
@@ -228,17 +228,18 @@ \subsection{Architecture and Design}
...
 Ideally any scripting made available to users should use the Python language.
-Python is becoming the standard language for data science and as such has a wide range of libraries, documentation and resources available for it.
+Python is becoming the standard language for data science and as such has a wide range of libraries, documentation and resources available for it \cite{python-nature}, with some of the most commonly used organized under the SciPy.org umbrella \cite{scipy}.
+Python also works fluently with the Jupyter electronic notebook, an increasingly popular tool in the scientific community \cite{jupyter-nature} and frequently used for tutorials.
 
 If the application requires a graphical user interface (GUI) then a good choice of library is Qt \cite{Qt}.
-Qt is an application framework commonly used by many open source and commercial software packages.
+Qt is an application framework commonly used by many open source and commercial software packages and is also a popular framework for GUI development in the neutron scattering and muon spectroscopy community (c.f., Section \ref{SINE2020}.
```
Comment (DanNixon) [inline guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex]:
```suggestion
Qt is an application framework commonly used by many open source and commercial software packages.
```

This is irrelevant here.

File: guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex [hunk 2]
```diff
@@ -228,17 +228,18 @@ \subsection{Architecture and Design}
...
 Each module should perform a very specific task (e.g. minimization) and should only contain the relevant functionality.
 Code within a module should be closely related, reducing the need to import unnecessary functionality.
 It is possible to test each module in isolation using automated testing (i.e. unit testing), reducing the risk of bugs being introduced into the code base.
 
 It can be desirable to allow users to write scripts to further extend the functionality of the software.
 Ideally any scripting made available to users should use the Python language.
-Python is becoming the standard language for data science and as such has a wide range of libraries, documentation and resources available for it.
+Python is becoming the standard language for data science and as such has a wide range of libraries, documentation and resources available for it \cite{python-nature}, with some of the most commonly used organized under the SciPy.org umbrella \cite{scipy}.
+Python also works fluently with the Jupyter electronic notebook, an increasingly popular tool in the scientific community \cite{jupyter-nature} and frequently used for tutorials.
```
Comment (DanNixon) [inline guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex]:
```suggestion
Python also works fluently with the Jupyter notebook, an increasingly popular tool in the scientific community \cite{jupyter-nature}.
```

File: guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex [hunk 3]
```diff
@@ -228,17 +228,18 @@ \subsection{Architecture and Design}
...
-Care should be taken when designing these module.
+Care should be taken when designing these modules.
 Each module should perform a very specific task (e.g. minimization) and should only contain the relevant functionality.
 Code within a module should be closely related, reducing the need to import unnecessary functionality.
 It is possible to test each module in isolation using automated testing (i.e. unit testing), reducing the risk of bugs being introduced into the code base.
 
 It can be desirable to allow users to write scripts to further extend the functionality of the software.
 Ideally any scripting made available to users should use the Python language.
-Python is becoming the standard language for data science and as such has a wide range of libraries, documentation and resources available for it.
+Python is becoming the standard language for data science and as such has a wide range of libraries, documentation and resources available for it \cite{python-nature}, with some of the most commonly used organized under the SciPy.org umbrella \cite{scipy}.
```
Comment (AnthonyLim23) [inline guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex]:
```suggestion
Python is becoming the standard language for data science and as such has a wide range of libraries, documentation and resources are available for it \cite{python-nature}.
Some of the most commonly used libraries are linked from SciPy.org \cite{scipy}.
```

File: guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex (hunk 1 above)
Comment (AnthonyLim23) [reply guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex]: I dont think it adds anything to say we use it in neutron and muon software. It is included in the open source part

Comment (thomasholmrod) [reply guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex]: ok

File: guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex (hunk 2 above)
Comment (thomasholmrod) [reply guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex]: ok

File: guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex (hunk 3 above)
Comment (thomasholmrod) [reply guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex]: ok

**Pull request merged**